sdk/eventhub/azure-eventhub/azure/eventhub/_utils.py

```diff
@@ -254,3 +261,37 @@ def parse_sas_credential(credential):
         if item.startswith('se='):
             expiry = int(item[3:])
     return (sas, expiry)
+
+def _convert_to_single_event_data(message, message_type):
+    # type: (Union[AmqpAnnotatedMessage, EventData], Type[EventData]) -> EventData
+    try:
+        # EventData
+        # pylint: disable=protected-access
+        return message._to_outgoing_message()  # type: ignore
+    except TypeError:
+        # AmqpAnnotatedMessage
+        # pylint: disable=protected-access
+        return message_type._from_message(message.to_outgoing_amqp_message())  # type: ignore
+    # best way to do below? is it needed?
+    #except:
+    #    raise TypeError(
+    #        "Only AmqpAnnotatedMessage or EventData instances are supported. "
+    #        "Received instead: {}".format(message.__class__.__name__)
+    #    )
```

Comment:
what's the current behavior if an object of non-EventData type is added to a batch?
wondering whether raising TypeError would be a breaking change.

Reply:
It results in an AttributeError. Since the exception above with AmqpAnnotatedMessage also uses `AttributeError`, would it be better to remove this? Or keep and `raise AttributeError("Only Amqp...")`?